### ISSUE TITLE
firewallcmd-ipset: Set cannot be created

### ISSUE THREAD

**Issue by acsfer** [fail2ban/fail2ban#2963]
```
fail2ban-client version
0.10.6
```
When `banaction = firewallcmd-ipset` i get this at every ban:

```
2021-03-08 13:53:30,021 fail2ban.actions        [1581]: NOTICE  [wordpress-hard] Ban 61.6.230.199
2021-03-08 13:53:34,659 fail2ban.utils          [1581]: ERROR   7f0c28724510 -- exec: ipset create f2b-wordpress-hard hash:ip timeout 0
firewall-cmd --direct --add-rule ipv4 filter INPUT_direct 0 -p tcp -m multiport --dports "$(echo '1:65535' | sed s/:/-/g)" -m set --match-set f2b-wordpress-hard src -j REJECT --reject-with icmp-port-unreachable
2021-03-08 13:53:34,660 fail2ban.utils          [1581]: ERROR   7f0c28724510 -- stderr: 'ipset v6.34: Kernel support protocol versions 6-7 while userspace supports protocol versions 6-6'
2021-03-08 13:53:34,660 fail2ban.utils          [1581]: ERROR   7f0c28724510 -- stderr: 'Set cannot be created: set with the same name already exists'
2021-03-08 13:53:34,660 fail2ban.utils          [1581]: ERROR   7f0c28724510 -- stderr: 'Error: COMMAND_FAILED'
2021-03-08 13:53:34,660 fail2ban.utils          [1581]: ERROR   7f0c28724510 -- returned 13
2021-03-08 13:53:34,660 fail2ban.actions        [1581]: ERROR   Failed to execute ban jail 'wordpress-hard' action 'firewallcmd-ipset' info 'ActionInfo({'ip': '61.6.230.199', 'family': 'inet4', 'fid': <function Actions.ActionInfo.<lambda> at 0x7f0c31631950>, 'raw-ticket': <function Actions.ActionInfo.<lambda> at 0x7f0c31631ea0>})': Error starting action Jail('wordpress-hard')/firewallcmd-ipset: 'Script error'
```

**Comment by sebres** [comment]
If you don't have some misconfiguration, this looks either like an aftereffect - for some reason the ipset of fail2ban jail with name `f2b-wordpress-hard` already exists.
This can have several reasons:
- at some time point related firewalld (or iptables) rule gets flushed, but ipset is still remaining, so an attempt to start or repair jail fails.
- the ipset is remaining after unsuccessful shutdown of fail2ban or we have indeed some error in start or stop of action `firewallcmd-ipset`
- there is some action definition for other jail (e. g. copy&paste) where name of the ipset in action (or name of jail resulting to ipset name) defined as parameter for some reason, check `fail2ban-client -d | grep wordpress-hard` whether there were some possible duplicates by ipset creation rules.

It is hard to follow what is wrong here now (since as already said an aftereffect). 
Can you probably take a look for errors occurring immediately before this kind of errors occurs initially (e. g. from previous stop of fail2ban or implicit before this throwed a first time)?

To repair this one could delete ipset manually, but if it is indeed a misconfiguration, you would quasi unban several IPs (affecting another jail or its different banning action).

Just to link to #2620 (which theoretically could help to avoid inconsistency issues after "flush").

**Comment by acsfer** [comment]
Sorry about the inactivity, but the impacted service is in the OVH burned datacenter :)
Not a critical one and it shoul be up next week, so i can take a look.

**Comment by sebres** [comment]
OK, you can ping me if the issue is still acute (or you need some help), I'll reopen it then.

**Comment by piwawa** [comment]
> OK, you can ping me if the issue is still acute (or you need some help), I'll reopen it then.

```
2023-07-22 14:42:41,741 fail2ban.actions        [14472]: NOTICE  [frps-ssh-ban] Restore Ban 117.141.150.30
2023-07-22 14:42:42,053 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- exec: ipset create f2b-frps-ssh-ban hash:ip timeout 0 
firewall-cmd --direct --add-rule ipv4 filter all 0 -p tcp -m multiport --dports "$(echo 'all' | sed s/:/-/g)" -m set --match-set f2b-frps-ssh-ban src -j REJECT --reject-with icmp-port-unreachable
2023-07-22 14:42:42,053 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- stderr: 'ipset v7.1: Set cannot be created: set with the same name already exists'
2023-07-22 14:42:42,054 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- stderr: "Error: COMMAND_FAILED: '/usr/sbin/iptables-restore -w -n' failed: iptables-restore v1.4.21: invalid port/service `all' specified"
2023-07-22 14:42:42,054 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- stderr: 'Error occurred at line: 2'
2023-07-22 14:42:42,054 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- stderr: "Try `iptables-restore -h' or 'iptables-restore --help' for more information."
2023-07-22 14:42:42,054 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- stderr: ''
2023-07-22 14:42:42,054 fail2ban.utils          [14472]: ERROR   7f97e5caeed0 -- returned 13
2023-07-22 14:42:42,054 fail2ban.actions        [14472]: ERROR   Failed to execute ban jail 'frps-ssh-ban' action 'firewallcmd-ipset' info 'ActionInfo({'ip': '117.141.150.30', 'fid': <function <lambda> at 0x7f97e4b1e1b8>, 'family': 'inet4', 'raw-ticket': <function <lambda> at 0x7f97e4b1e758>})': Error starting action Jail('frps-ssh-ban')/firewallcmd-ipset: 'Script error'
```

I am experiencing this problem too, how to solve this??

**Comment by sebres** [comment]
> I am experiencing this problem too...

Nope. This is another problem.
It looks like for jail frps-ssh-ban either no ports specified (e.g. `all` used somewhere), or (if wanted) one'd need to use different banaction to ban all ports. For example like here:
https://github.com/fail2ban/fail2ban/blob/eea1881b734b73599a21df2bfbe58b11f78d0a46/config/jail.conf#L827
Or if appropriate default is not set, try to set it directly:
```ini
[frps-ssh-ban]
banaction = firewallcmd-ipset[actiontype=allports]
```

**Comment by piwawa** [comment]
> 

Here is my default setting, i just follow a tutorial and write this, but don't know why...
```
[DEFAULT]
ignoreip = 127.0.0.1/8
bantime = 86400
findtime = 300
maxretry = 5
backend = auto
banaction = firewallcmd-ipset
action = %(action_mwl)s

# 防止SSH爆破
[sshd]
enabled = true
filter = sshd 
logpath = /var/log/secure
findtime = 300
maxretry = 5
port = 22
action = %(action_mwl)s

# FRP SSH防爆破
[frps-ssh-ban]
enabled = true
findtime = 3m
maxretry = 3
filter = frps-ssh-ban
logpath = /root/frp_0.48.0/frps.log

protocol = all
port = all
chain = all

action = %(action_mwl)s
```
I am using `firewalld` instead of `iptables`. Is there something wrong to set?
